<!DOCTYPE html>
<html lang="en-us">
<head>
    <meta charset="UTF-8">
    <title>Computational Biology by computational-bio</title>
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <link rel="stylesheet" type="text/css" href="stylesheets/normalize.css" media="screen">
    <link rel="stylesheet" type="text/css" href="stylesheets/bootstrap.css" media="screen">
    <link rel="stylesheet" type="text/css" href="stylesheets/bootstrap-theme.min.css" media="screen">
    <link href='https://fonts.googleapis.com/css?family=Open+Sans:400,700' rel='stylesheet' type='text/css'>
    <link rel="stylesheet" type="text/css" href="stylesheets/stylesheet.css" media="screen">
    <link rel="stylesheet" type="text/css" href="stylesheets/github-light.css" media="screen">
    <script type="text/javascript" src="js/jquery-2.2.0.min.js"></script>
    <script type="text/javascript" src="js/bootstrap.min.js"></script>
    <script type="text/javascript" src="js/npm.js"></script>

    <script type="text/javascript">

        $(document).ready(function () {
        });

    </script>
</head>
<body>
<section class="page-header">
    <h1 class="project-name">Computational Biology</h1>
    <h2 class="project-tagline">Computational Biology Project</h2>
</section>

<section class="main-content">
    <!--<h1 style="text-align: center;">
        <a id="coming-soon" class="anchor" href="#coming-soon" aria-hidden="true"><span aria-hidden="true"
                                                                                        class="octicon octicon-link"></span></a><strong>Coming
        Soon</strong>
    </h1>-->

    <div>
        <!-- Nav tabs -->
        <ul class="nav nav-tabs" role="tablist" id="myTabs-main">
            <li role="presentation" class="active"><a href="#sequence-alignment" aria-controls="sequence-alignment"
                                                      role="tab" data-toggle="tab">Sequence Alignment</a></li>
            <li role="presentation"><a href="#paper-review" aria-controls="paper-review" role="tab" data-toggle="tab">Paper
                Review</a></li>
            <li role="presentation"><a href="#project-proposal" aria-controls="project-proposal" role="tab"
                                       data-toggle="tab">Project Proposal</a></li>
            <li role="presentation"><a href="#project-update" aria-controls="project-update" role="tab"
                                       data-toggle="tab">Project</a></li>
        </ul>

        <!-- Tab panes -->
        <div class="tab-content">
            <div role="tabpanel" class="tab-pane fade in active" id="sequence-alignment">
                <h1 style="text-align: center; padding-top: 10px;">
                    <a id="motivation" class="anchor" href="#motivation" aria-hidden="true">
                        <span aria-hidden="true" class="octicon octicon-link"></span></a>
                    <strong>Motivation</strong>
                </h1>

                <p class="text-style">Aligning nucleotide or amino acid sequences is a very common procedure in HIV
                    computational

                    analysis. An accurate alignment is the first step in making a proper and correct analysis of HIV
                    datasets.

                    Sequence alignments are essential for phylogenetic analysis tracing the epidemiology of HIV, but
                    also

                    for interpretations of drug resistance and data mining efforts, where correctly positioning
                    nucleotides or

                    amino acids of different strains with respect to each other is pivotal.</p>

                <hr>

                <h1 style="text-align: center;">
                    <a id="objective" class="anchor" href="#objective" aria-hidden="true">
                        <span aria-hidden="true" class="octicon octicon-link"></span></a>
                    <strong>Objective</strong>
                </h1>

                <p class="text-style">The strains of HIV-1 can be classified into three groups: the "major" group M, the
                    "outlier"
                    group O and

                    the "new" group N. More than 90% of HIV-1 infections belong to HIV-1 group M. Group M there are

                    known to be at least nine genetically distinct subtypes (or clades) of HIV-1. These are subtypes A,
                    B, C, D,

                    F, G, H, J and K.

                    <br><br>Our objective here is to compare <b>HIV-1 A</b> and <b>HIV-1 B</b> virus by determining the
                    similarity
                    between
                    the

                    two.</p>

                <div>
                    <img src="images/pic1.jpg">
                </div>

                <hr>

                <h1 style="text-align: center;">
                    <a id="method" class="anchor" href="#method" aria-hidden="true">
                        <span aria-hidden="true" class="octicon octicon-link"></span></a>
                    <strong>Method</strong>
                </h1>

                <p class="text-style">We selected two sequence of HIV-1 virus, <b>A</b> and <b>B</b> from HIV Sequence
                    Database.</p>

                <br>
                <div>
                    <!-- Nav tabs -->
                    <ul class="nav nav-tabs" role="tablist" id="myTabs">
                        <li role="presentation" class="active"><a href="#hiv-1A" aria-controls="hiv-1A" role="tab"
                                                                  data-toggle="tab">HIV-1 A</a></li>
                        <li role="presentation"><a href="#hiv-1B" aria-controls="hiv-1B" role="tab" data-toggle="tab">HIV-1
                            B</a></li>
                    </ul>

                    <!-- Tab panes -->
                    <div class="tab-content">
                        <div role="tabpanel" class="tab-pane fade in active" id="hiv-1A">
                            <p class="hiv-sequence">
                                TGGAAGGGTTAATTTACTCCAGGAGAAGACAAGAAATCCTTGATCTGTGGGTCTACCACACACAAGGCTTCTTCCCTGATTGGCAGGAATACACACCAGGGCCAGGGGTTAGATACCCATTAACATTTGGATGGTGCTTCAAGCTAGTACCAGTTGATCCAGATGAGGTAGAGGGGGCTACTGAGGGAGAGAACAACAGCCTGTTACACCCGATATGTCAACATGGAATGGATGATGAGGAGAAAGAAGTATTAAAGTGGAAGTTTGACAGCCGCCTGGCACTAAAACACAGAGCCCAAGAGATGCATCCGGAGTTCTACAAGAACTGAAGAACTGCTGACACAGAAGTTGCTGACAGGGACTTTCTGATGGGGACTTTCCAGGGGAGGTGTGGTTTGGGCGGAGTTGGGGAGTGGCTAACCCTCAGATGCTGCATATAAGCAGCTGCTTCTCGCTTGTACTGGGTCTCTCTTGTTAGACCAGATCGAGCCTGGGAGCTCTCTGGCTAGCTAGGGAACCCACTGCTTAAGCCTCAATAAAGCTTGCCTTGAGTGCTTCAAGTAGTGTGTGCCCATCTGTTGTGACTCTGGTAACTAGAGATCCCTCAGACCTCTTTAGTCAGTGTGGAAAATCTCTAGCA
                            </p>
                        </div>
                        <div role="tabpanel" class="tab-pane fade" id="hiv-1B">
                            <p class="hiv-sequence">
                                TGGATGGGCTAATTTACTCCCAAAAAAGACAAGATATCCTTGATTTGTGGGTCTACCACACACAAGGCTACTTCCCTGATTGGCAGAACTACACACCAGGGCCAGGGACCAGATTTCCACTGACCTTTGGATGGTGCTTCAAGCTAGTACCAGTTGAGCCAGAGAAGGTAGAGGAGGCTAATGAAGGAGAGAACAACTGCTTGTTACACCCTATGAGCCTGCATGGGATGGAGGACCCGGAGAAAGAAGTGTTAGTATGGAAGTTTGACAGCACCCTAGCATTCCATCACGTGGCCCGAGAGCTGCATCCGGAGTACTACAGAGACTGCTGACATTGAGCTTTCTACAAGGGACTTTCCGCTGGGGACTTTCCAGGGAGGCGTGGCCTGGGCGGGACTGGGGAGTGGCGAGCCCTCAGATGCTGCATATAAGCAGCTGCTTTTTGCCTGTACTGGGTCTCTCTGGTTAGACCAGATCTGAGCCTGGGAGCTCTCTGGCTAACTAGGGAACCCACTGCTTAAGCCTCAATAAAGCTTGCCTTGAGTGCTTCAAGTAGTGTGTGCCCGTCTGTTGTGTGACTCTGGTAACTAGAGATCCCTCAGACCCTTTTAGTCAGTGTGGAAAATCTCTAGCA
                            </p>
                        </div>
                    </div>

                </div>
                <br><br>

                <p class="text-style">
                    We compared the DNA strands of HIV-1A and HIV-1B using two different techniques –
                </p>

                <div class="panel-group" id="accordion" role="tablist" aria-multiselectable="true">
                    <div class="panel panel-default">
                        <div class="panel-heading" role="tab" id="headingOne">
                            <h4 class="panel-title">
                                <a role="button" data-toggle="collapse" data-parent="#accordion" href="#collapseOne"
                                   aria-expanded="true" aria-controls="collapseOne">
                                    1. Multiple Sequence Alignment (MSA)
                                </a>
                            </h4>
                        </div>
                        <div id="collapseOne" class="panel-collapse collapse in" role="tabpanel"
                             aria-labelledby="headingOne">
                            <div class="panel-body">
                                <b> Multiple Sequence Alignment (MSA)</b> is generally the alignment of three or more
                                biological

                                sequences (protein or nucleic acid) of similar length. From the output, homology can be
                                inferred

                                and the evolutionary relationships between the sequences studied.

                                <p>Tool :
                                    <a href="http://www.ebi.ac.uk/Tools/msa/clustalo/">Cluster Omega</a>
                                </p>
                                <hr>
                                <h3 style="text-align: center;">
                                    <a id="output-1" class="anchor" href="#output" aria-hidden="true">
                                        <span aria-hidden="true" class="octicon octicon-link"></span></a>
                                    <strong>Output</strong>
                                </h3>
                                <div>
                                    <h4>Sequence Alignment using cluster omega</h4>
                                    <img src="images/Omega1.jpg">
                                </div>
                                <div class="row">
                                    <div class="col-md-6">
                                        <table class="table-striped">
                                            <tbody>
                                            <tr>
                                                <td>Gaps</td>
                                                <td>12</td>
                                            </tr>
                                            <tr>
                                                <td>Mismatches</td>
                                                <td>87</td>
                                            </tr>
                                            <tr>
                                                <td>Similarity</td>
                                                <td>544</td>
                                            </tr>
                                            </tbody>
                                        </table>
                                    </div>
                                    <div class="col-md-6">
                                        <img src="images/Omega-2.jpg">
                                    </div>
                                </div>
                            </div>
                        </div>
                    </div>
                    <div class="panel panel-default">
                        <div class="panel-heading" role="tab" id="headingTwo">
                            <h4 class="panel-title">
                                <a class="collapsed" role="button" data-toggle="collapse" data-parent="#accordion"
                                   href="#collapseTwo" aria-expanded="false" aria-controls="collapseTwo">
                                    2. Pair-wise sequence Alignment
                                </a>
                            </h4>
                        </div>
                        <div id="collapseTwo" class="panel-collapse collapse" role="tabpanel"
                             aria-labelledby="headingTwo">
                            <div class="panel-body">
                                <b>Pair-wise sequence Alignment</b> is used to identify regions of similarity that may
                                indicate

                                functional, structural and/or evolutionary relationships between two biological
                                sequences

                                (protein or nucleic acid).
                                <p>Tool :
                                    <a href="http://www.ebi.ac.uk/Tools/psa/emboss_needle/nucleotide.html">Needle</a>
                                </p>
                                <hr>
                                <h3 style="text-align: center;">
                                    <a id="output-2" class="anchor" href="#output" aria-hidden="true">
                                        <span aria-hidden="true" class="octicon octicon-link"></span></a>
                                    <strong>Output</strong>
                                </h3>
                                <div>
                                    <h4>Pairwise Alignment using needle</h4>
                                    <img src="images/Needle2.jpg">
                                    <img src="images/Needle1.jpg">
                                </div>
                            </div>
                        </div>
                    </div>
                </div>
                <br>

                <hr>

                <div style="clear: both;"></div>
                <br>

                <h1 style="text-align: center;">
                    <a id="conclusion" class="anchor" href="#conclusion" aria-hidden="true">
                        <span aria-hidden="true" class="octicon octicon-link"></span></a>
                    <strong>Conclusion</strong>
                </h1>

                <p class="text-style">
                    Even though HIV1A and B originate from different regions viz. Subtype A is common in West Africa
                    while Subtype B
                    is the dominant form in Europe, the Americas, Japan, Thailand, and Australia, The sequence alignment
                    results
                    show a similarity of <b>85-86%</b> position with <b>12-16</b> gaps and <b>~82-87</b> mismatches.
                </p>

                <hr>

                <!--<h1 style="text-align: center;">
                    <a id="future-work" class="anchor" href="#future-work" aria-hidden="true">
                        <span aria-hidden="true" class="octicon octicon-link"></span></a>
                    <strong>Future Work</strong>
                </h1>

                <hr>-->

                <h1 style="text-align: center;">
                    <a id="bibliography" class="anchor" href="#bibliography" aria-hidden="true">
                        <span aria-hidden="true" class="octicon octicon-link"></span></a>
                    <strong>Bibliography</strong>
                </h1>

                <ul class="text-style">
                    <li>Ana Abecasis, Anne-Mieke Vandamme, and Philippe Lemey. Sequence Alignment in HIV Computational
                        Analysis.
                    </li>
                    <li>Melsted, Páll, and Jonathan K. Pritchard. Efficient Counting of K-mers in DNA Sequences Using a
                        Bloom Filter. <i>BMC Bioinformatics 12.1</i> (2011): 333. Web.
                    </li>
                </ul>

                <hr>

                <h2 style="text-align: center;">
                    <a id="team" class="anchor" href="#team" aria-hidden="true">
                        <span aria-hidden="true" class="octicon octicon-link"></span></a>
                    Contributors
                </h2>

                <div class="contributors">
                    <div>
                        <a href="https://www.linkedin.com/in/ashwinbansod" class="user-mention">
                            <img src="images/Ashwin.png">
                            <br>Ashwin Bansod</a>
                        <!--<p class="contributors-description">
                            Computer Science graduate student at University of Illinois at Chicago with over 3.8 years of experience
                            in
                            software development, product release, requirement gathering, client interaction. Experience in
                            financial
                            services, stock markets developing low latency high throughput trading system.-->
                    </div>
                    <div>
                        <a href="https://www.linkedin.com/in/krutarthjoshi18" class="user-mention">
                            <img src="images/Krutarth.png">
                            <br>Krutarth Joshi</a>
                        <!--<p class="contributors-description">
                            Pursuing Masters of Science in Computer Science at University of Illinois at Chicago with specialization
                            in
                            Machine Learning and Data Science.
                            Researched various supervised machine learning techniques, and applied them to
                            create an alarm generation system to detect and hence, predict and prevent bearing failures in
                            aero-derivative engines.-->
                    </div>
                    <div>
                        <a href="https://www.linkedin.com/in/kushagrathapar" class="user-mention">
                            <img src="images/Kushagra.png">
                            <br>Kushagra Thapar</a>
                        <!--<p class="contributors-description">
                            Software Developer with 3 years of experience in web application development, web services, database
                            designs, designing product architecture from scratch.
                            Currently pursuing Masters of Science in Computer Science from University of Illinois at Chicago with
                            specialization in Artificial Intelligence-->
                    </div>
                </div>
            </div>
            <div role="tabpanel" class="tab-pane fade" id="paper-review">
                <h2 style="text-align: center; padding-top: 20px">An Exact Algorithm to Compute the DCJ Distance for
                    Genomes with Duplicate Genes</h2>
                <h3 style="text-align: center">
                    Mingfu Shao, Yu Lin, and Bernard Moret
                    <h4 style="text-align: center; margin-top: 1em">
                        Laboratory for Computational Biology and Bioinformatics, EPFL, Lausanne, Switzerland<br>
                        Department of Computer Science and Engineering, University of California, San Diego,
                        La Jolla, California<br>
                        {mingfu.shao,yu.lin,bernard.moret}@epfl.ch
                    </h4>
                </h3>
                <p class="text-style">
                    <br>The double-cut-and-join (DCJ) model has been used to calculate edit distance between two genomes
                    for quite a long time. Optimizing edit distance problem forms an integral part to solve basic
                    problems for genome evolution. Shao, Lin, and Moret proposes an Integer Linear Programming approach
                    to tackle a specific case of genomes i.e. genomes with duplicate genes. This problem can be solved
                    in linear time for genomes without duplicate genes but has been proved NP-Hard for genomes with
                    duplicate genes by a reduction from NP-Hard problem of Breakpoint Graph Decomposition.
                    <br><br>The authors define this problem as to find a valid bijection between two genomes with
                    duplicate gene content that minimizes the DCJ distance between them. When there is a consistent
                    decomposition with <i>c</i> cycles and <i>o</i> odd-length paths, DCJ operations needed to transform
                    genome G1 to G2 is given by <i>(|V|/4 − c − o/2)</i>. Hence to minimize total DCJ operations,
                    maximize <i>(c + o/2)</i> over the space of all consistent decompositions. This reduces the problem
                    to <i>maximum cycle decomposition problem</i> which is further formulated as an Integer Linear
                    Program. The ILP formulation has O|E| variables, O|E| constraints and maximizes the number of
                    cycles.
                    <br><br>Authors have devised a preprocessing algorithm to reduce complexity while preserving
                    optimality of DCJ distance computing algorithm. The combined approach is crucial for comparing
                    genomes with duplicate genes as they are commonly observed in most species.
                    <br><br>Authors claim a bold assumption that after a speciation event, only DCJ operations are
                    involved. This is unrealistic and is only used to simplify the problem.
                </p>
            </div>
            <div role="tabpanel" class="tab-pane fade" id="project-proposal">
                <br><br>
                <p>Kinship inference of a population is an indispensable component of understanding genetic relatedness
                    or
                    coefficient of relationship. It helps biologists to understand more profoundly how traits inherit
                    from
                    parents to their offspring. It also includes study of mating process so as to correctly construct
                    relationship between a parent and its offspring. One of the major steps in kinship analysis is the
                    formalization of biological problems into computational problems which can then be solved using
                    statistical or combinatorial approaches [1].</p>
                <p>Various biological problems pertaining to kinship inference have already been formalized including
                    but
                    not limited to MIN-PARENT Problem [2], MIN MATING Problem and Parentage assignment problem.</p>
                <p>Formalization of these problems have been very helpful to biologists for analyzing consanguinity and
                    evolution of genetic traits among different generations. We have not been able to completely analyze
                    these processes by only finding the minimum number of parents. There are many variances of these
                    problems which are yet to be formalized and classified into easy and hard categories.</p>
                <p>We aim to formalize different variances of these problems using various constraints and restrictions.
                    Our focus is mainly to formalize Full Sibling Reconstruction and Parentage Assignment Problem. We
                    propose to define and impose a set of constraints in order to generate different variances of the
                    above
                    problems. For example, Full Sibling Reconstruction problem can be constrained on the mating process,
                    whereas Parentage Assignment Problem’s variances can be constructed on the basis of unbalanced sex
                    ratios or minimizing the number of females. We are also aiming to classify these problems into easy
                    or
                    hard categories accordingly.</p>
                <p>Investigation of many fundamental biological phenomena including mating systems, selection and
                    adaptation, kin selection, and dispersal patterns depends on the growing development and application
                    of
                    molecular markers. The power and potential of the genotypic information obtained in these studies
                    often
                    rests in our ability to reconstruct genealogical relationships among individuals. Formalization of
                    these
                    problems will prove to be a great advantage for above studies.
                <h3 style="text-align: center">Bibliography</h3>
                <p>[1] Mary V. Ashley, Tanya Y. Berger-Wolf, Isabel C. Caballero, Wanpracha Chaovalitwongse,
                    Bhaskar DasGupta, and Saad I. Sheikh. Full Sibling Reconstruction in Wild Populations from
                    Microsatellite Genetic Markers.</p>

                <p>[2] Ashley, Mary V., Tanya Y. Berger-Wolf, Wanpracha Chaovalitwongse, Bhaskar Dasgupta, Ashfaq
                    Khokhar, and Saad Sheikh. On Approximating an Implicit Cover Problem in Biology. <i>Algorithmic
                        Aspects in Information and Management Lecture Notes in Computer Science (2009)</i>: 43-54. Web.
                </p>

                <hr>
                <p class="text-style">
                    <a href="project-proposal/Comp-Bio-Proposal-revisited.pdf" download="Project-Proposal">Download
                        Proposal</a>
                </p>
            </div>
            <div role="tabpanel" class="tab-pane fade" id="project-update">
                <h2 style="text-align: center; padding-top: 20px">Formalizing Biological Problems for Kinship
                    Inference</h2>
                <h3 style="text-align: center; margin-top: 1em">Introduction and Motivation</h3>
                <p class="text-style">Kinship inference of a population is an indispensable component of understanding
                    genetic relatedness or coefficient of relationship. A natural field biology question is to
                    characterize genealogical relationships in wild populations, such as which organisms are siblings.
                    Biologists want to know how individuals survive, acquire mates, reproduce, and disperse to new
                    populations. It helps biologists to understand more profoundly how traits inherit from parents to
                    their offspring. It also includes study of mating process so as to correctly construct relationship
                    between a parent and its offspring. One of the major steps in kinship analysis is the formalization
                    of biological problems into computational problems which can then be solved using statistical or
                    combinatorial approaches.
                </p>
                <p class="text-style">This project presents generalized formalizations of biological problems. In this
                    project we are only
                    concerned with full sibling relationship from single generation sample of microsatellite markers.
                    The formalizations defined use the Mendelian inheritance rules to impose constraints on genetic
                    content possibilities of a sibling group. The formalizations of the inferred constraints under the
                    parsimonious assumption of minimum number of parents leads to minimum MIN-PARENT problem. Variations
                    of these MIN-PARENT problems are defined in this project.</p>
                <h3 style="text-align: center; margin-top: 1em">Formulations</h3>
                <h4><b>Sibling Reconstruction Problem</b></h4>
                <p class="text-style">Given a genetic (microsatellite) sample from a population of <i>n</i> diploid
                    individuals of the same generation, <i>U</i>, the goal is to reconstruct the full sibling groups
                    (groups of individuals with the same parents). We assume no knowledge of parental information.</p>
                <p>Formally, we are given a set <i>U</i> of <i>n</i> individual microsatellite samples from <i>l</i>
                    genetic loci</p>
                <div style="text-align: center">
                    <img src="images/sibship-1.png" width="500em">
                </div>
                and a<sub>ij</sub> and b<sub>ij</sub> are the two alleles of the diploid individual <i>i</i> at locus
                <i>j</i>. The goal is to find a partition of individuals P<sub>1</sub>, ... P<sub>m</sub> such that
                <div style="text-align: center">
                    <img src="images/sibship-2.png" width="500em">
                </div>
                Notice, here that we have not defined the function <b><i>Parents(x)</i></b>. This is a biological
                objective.
                <h4><b>MIN-PARENT Problem</b></h4>
                <p class="text-style">Given the Mendelian constraints, our goal is to cover the Universe <i>U</i> by a
                    set of full-sibling groups under the parsimonious assumption of a minimum number of parents.
                    Formally, the MIN-PARENT problem is defined as follows:</p>
                <div style="text-align: left">
                    <img src="images/min-parent-1.png" width="600em" style="padding-left: 10px;">
                    <img src="images/min-parent-2.png" width="400em" style="padding-left: 15px;">
                </div>
                <h4 style="color: black;"><b>Standard Terminologies</b></h4>
                <div style="text-align: left">
                    <img src="images/standard-terminology.png" width="550em" style="padding-left: 20px;">
                </div>
                <p>Assuming, our polynomial time oracle <i>O</i> answers following queries:
                <ul>
                    <li>
                        Given a subset <i>U<sup>'</sup></i> of Universe <i>U</i>, does <i>U<sup>'</sup></i> form a valid
                        sibling set following Mendelian constraints.
                    </li>
                </ul>
                </p>
                <p>Below are the formulations of variations of MIN-PARENT problem: </p>
                <div class="panel-group" id="project-accordion" role="tablist" aria-multiselectable="true">
                    <div class="panel panel-default">
                        <div class="panel-heading" role="tab" id="problem-1">
                            <h4 class="panel-title">
                                <a role="button" data-toggle="collapse" data-parent="#project-accordion"
                                   href="#collapse-1"
                                   aria-expanded="true" aria-controls="collapse-1">
                                    1. FIND-MIN-PARENT MONOGAMY
                                </a>
                            </h4>
                        </div>
                        <div id="collapse-1" class="panel-collapse collapse in" role="tabpanel"
                             aria-labelledby="problem-1">
                            <div class="panel-body">
                                <b style="color: black;">Definition:</b>
                                <p>The goal of this problem is to find a minimum set of parents for a given sibling
                                    partition of the universe and a candidate set of parents having monogamous mating
                                    process.</p>
                                <b style="color: black;">Input:</b>
                                <ul>
                                    <li>
                                        Partition <i>A</i>
                                    </li>
                                    <li>
                                        Set of possible parents <i>P</i>
                                    </li>
                                    <li>
                                        Universe <i>U</i>
                                    </li>
                                </ul>
                                <b style="color: black;">Constraints:</b>
                                <ul>
                                    <li>Monogamous Mating Process</li>
                                </ul>
                                <b style="color: black;">Formulation:</b>
                                <div style="text-align: center">
                                    <img src="images/min-parent-monogamy-1.png" width="350em">
                                </div>
                                <b style="color: black;">Relation with Sibling Reconstruction:</b>
                                <p class="text-style" style="margin-bottom: 0;">Under monogamous constraints, number of
                                    minimum set parents set
                                    will always be equal to minimum sibling sub groups.</p>
                                <div style="text-align: center">
                                    <img src="images/min-parent-monogamy-2.png" width="120em">
                                </div>
                                <p>Hence, these problems are equivalent, but not vice-versa. We prove this in the
                                    following lemma.</p>
                                <br>
                                <b style="color: black;">Lemma:</b> <b>If minimum number of matings is equal to the
                                number of full-sibling sub groups, then it does not necessary imply monogamy.</b>
                                <br><br>
                                <p><b style="color: black;">Proof:</b> Suppose we have a set of parents <i>P</i>,
                                    <br>
                                <div style="text-align: center">P = {A, B, C, D, E}</div>
                                <br>
                                And a partition set <i>A</i>, for which each set <i>S</i> in <i>A</i> is a sibling set.
                                <br><br>
                                <div style="text-align: center">A = {{S<sub>1</sub>}, {S<sub>2</sub>}, {S<sub>3</sub>}}
                                </div>
                                <br>
                                Using our oracle <i>O</i>, we find the parents of partition set <i>A</i>.
                                <p>We get the following results:</p>
                                <div style="text-align: center">
                                    B(S<sub>1</sub>) = (B, C)<br>
                                    B(S<sub>2</sub>) = (A, D)<br>
                                    B(S<sub>3</sub>) = (A, E)<br><br>
                                </div>
                                <p>Here we have 3 matings, which is equal to the number of full-sibling sub groups we
                                    originally have. And we can clearly observe that the mating process is not <i>Monogamous</i>
                                    as parent <i>'A'</i> is shared with 2 sibling sets, i.e. <i>S<sub>2</sub></i> and
                                    <i>S<sub>3</sub></i>.
                                </p>
                            </div>
                        </div>
                    </div>
                    <div class="panel panel-default">
                        <div class="panel-heading" role="tab" id="problem-2">
                            <h4 class="panel-title">
                                <a class="collapsed" role="button" data-toggle="collapse"
                                   data-parent="#project-accordion"
                                   href="#collapse-2" aria-expanded="false" aria-controls="collapse-2">
                                    2. FIND-MIN-PARENT POLYGYNY and POLYANDRY
                                </a>
                            </h4>
                        </div>
                        <div id="collapse-2" class="panel-collapse collapse" role="tabpanel"
                             aria-labelledby="problem-2">
                            <div class="panel-body">
                                <b style="color: black;">Definition:</b>
                                <p>The goal of this problem is to find a minimum set of parents for a given sibling
                                    partition of the universe and a candidate set of parents having polygynous or
                                    polyandrous mating process.</p>
                                <b style="color: black;">Input:</b>
                                <ul>
                                    <li>
                                        Partition <i>A</i>
                                    </li>
                                    <li>
                                        Set of possible parents <i>P</i>
                                    </li>
                                    <li>
                                        Universe <i>U</i>
                                    </li>
                                </ul>
                                <b style="color: black;">Constraints:</b>
                                <ul>
                                    <li>Polygynous and Polyandrous Mating Process</li>
                                </ul>
                                <b style="color: black;">Formulation:</b>
                                <div style="text-align: center">
                                    <img src="images/min-parent-polygyny-polyandry-1.png" width="380em">
                                </div>
                                <b style="color: black;">Relation with Sibling Reconstruction:</b>
                                <p class="text-style" style="margin-bottom: 0;">Under polygynous or polyandrous mating
                                    constraints, this problem is easy when compared to full-sibling reconstruction
                                    problem.</p>
                            </div>
                        </div>
                    </div>
                    <div class="panel panel-default">
                        <div class="panel-heading" role="tab" id="problem-3">
                            <h4 class="panel-title">
                                <a class="collapsed" role="button" data-toggle="collapse"
                                   data-parent="#project-accordion"
                                   href="#collapse-3" aria-expanded="false" aria-controls="collapse-3">
                                    3. FIND-MIN-PARENT PROMISCUOUS
                                </a>
                            </h4>
                        </div>
                        <div id="collapse-3" class="panel-collapse collapse" role="tabpanel"
                             aria-labelledby="problem-3">
                            <div class="panel-body">
                                <b style="color: black;">Definition:</b>
                                <p>The goal of this problem is to find a minimum set of parents for a given sibling
                                    partition of the universe and a candidate set of parents having promiscuous mating
                                    process.</p>
                                <b style="color: black;">Input:</b>
                                <ul>
                                    <li>
                                        Partition <i>A</i>
                                    </li>
                                    <li>
                                        Set of possible parents <i>P</i>
                                    </li>
                                    <li>
                                        Universe <i>U</i>
                                    </li>
                                </ul>
                                <b style="color: black;">Constraints:</b>
                                <ul>
                                    <li>Promiscuous Mating Process</li>
                                </ul>
                                <b style="color: black;">Valid Solutions:</b>
                                <div style="text-align: center">
                                    <img src="images/min-parent-promiscuous-1.png" width="90em">
                                </div>
                                <b style="color: black;">Formulation:</b>
                                <div style="text-align: center">
                                    <img src="images/min-parent-promiscuous-2.png" width="200em">
                                </div>
                                <b style="color: black;">Relation with Sibling Reconstruction:</b>
                                <p class="text-style" style="margin-bottom: 0;">Under promiscuous mating
                                    process, this problem is very hard when compared to full sibling reconstruction
                                    problem.</p>
                            </div>
                        </div>
                    </div>
                    <div class="panel panel-default">
                        <div class="panel-heading" role="tab" id="problem-4">
                            <h4 class="panel-title">
                                <a class="collapsed" role="button" data-toggle="collapse"
                                   data-parent="#project-accordion"
                                   href="#collapse-4" aria-expanded="false" aria-controls="collapse-4">
                                    4. MIN-FATHER POLYANDRY and MONOGAMY
                                </a>
                            </h4>
                        </div>
                        <div id="collapse-4" class="panel-collapse collapse" role="tabpanel"
                             aria-labelledby="problem-4">
                            <div class="panel-body">
                                <p>Depending upon the given input, this problem can be formulated in two different
                                    ways:</p>
                                <b style="color: black;">Definition - 1:</b>
                                <br>
                                <p>Given a set of mothers and their offspring, the goal of this problem is to minimize
                                    the number of fathers required to produce those offspring.</p>
                                <b style="color: black;">Input:</b>
                                <ul>
                                    <li>
                                        Set of possible fathers <i>W</i> = {w<sub>1</sub>, w<sub>2</sub>, w<sub>3</sub>
                                        ... }
                                    </li>
                                    <li>
                                        Set of mothers <i>V</i> = {v<sub>1</sub>, v<sub>2</sub>, v<sub>3</sub> ... }
                                    </li>
                                    <li>
                                        Set of offspring <i>S</i> = {s<sub>1</sub>, s<sub>2</sub>, s<sub>3</sub> ... }
                                    </li>
                                </ul>
                                <p>such that, <br>
                                    v<sub>1</sub> is the mother of offspring set s<sub>1</sub>,<br>
                                    v<sub>2</sub> is the mother of offspring set s<sub>2</sub>, and so on ...<br>
                                </p>
                                <b style="color: black;">Constraints:</b>
                                <ul>
                                    <li>Polyandrous and Monogamous Mating process</li>
                                </ul>
                                <b style="color: black;">Formulation:</b>
                                <div style="text-align: center">
                                    <img src="images/min-father-1.png" width="350em">
                                </div>

                                <hr>

                                <b style="color: black;">Definition - 2:</b>
                                <br>
                                <p>Given a set of offspring and a set of possible mothers, the goal of this problem is
                                    to minimize the number of parents required to produce those offspring.</p>
                                <b style="color: black;">Input:</b>
                                <ul>
                                    <li>
                                        Set of possible mothers <i>V</i> = {v<sub>1</sub>, v<sub>2</sub>, v<sub>3</sub>
                                        ... }
                                    </li>
                                    <li>
                                        Set of offspring <i>S</i> = {s<sub>1</sub>, s<sub>2</sub>, s<sub>3</sub> ... }
                                    </li>
                                </ul>
                                <b style="color: black;">Constraints:</b>
                                <ul>
                                    <li>Polyandrous and Monogamous Mating process</li>
                                </ul>
                                <b style="color: black;">Formulation:</b>
                                <div style="text-align: center">
                                    <img src="images/min-father-2.png" width="500em">
                                </div>
                                <hr>
                                <b style="color: black;">Relation with Sibling Reconstruction:</b>
                                <p class="text-style" style="margin-bottom: 0;">Under polyandrous or monogamous mating
                                    constraints, the minimum number of fathers required to produce these offspring will
                                    be equal to the number of full sibling sub groups.</p>
                                <div style="text-align: center">
                                    <img src="images/min-father-3.png" width="150em">
                                </div>
                                <br>
                                Hence, this problem is easy when compared to MIN-PARENT problem.
                            </div>
                        </div>
                    </div>
                    <div class="panel panel-default">
                        <div class="panel-heading" role="tab" id="problem-5">
                            <h4 class="panel-title">
                                <a class="collapsed" role="button" data-toggle="collapse"
                                   data-parent="#project-accordion"
                                   href="#collapse-5" aria-expanded="false" aria-controls="collapse-5">
                                    5. MIN-FATHER POLYGYNY
                                </a>
                            </h4>
                        </div>
                        <div id="collapse-5" class="panel-collapse collapse" role="tabpanel"
                             aria-labelledby="problem-5">
                            <div class="panel-body">
                                <b style="color: black;">Definition:</b>
                                <br>
                                <p>Given a set of offspring and a set of possible mothers, the goal of this problem is
                                    to minimize the number of parents required to produce those offspring.</p>
                                <b style="color: black;">Input:</b>
                                <ul>
                                    <li>
                                        Set of possible fathers <i>W</i> = {w<sub>1</sub>, w<sub>2</sub>, w<sub>3</sub>
                                        ... }
                                    </li>
                                    <li>
                                        Set of possible mothers <i>V</i> = {v<sub>1</sub>, v<sub>2</sub>, v<sub>3</sub>
                                        ... }
                                    </li>
                                    <li>
                                        Set of offspring <i>S</i> = {s<sub>1</sub>, s<sub>2</sub>, s<sub>3</sub> ... }
                                    </li>
                                </ul>
                                <b style="color: black;">Constraints:</b>
                                <ul>
                                    <li>Polygynous Mating process</li>
                                </ul>
                                <b style="color: black;">Formulation:</b>
                                <p>Since, the mating process is polygynous, therefore the formulation of this problem
                                    would be just to minimize the number of fathers required to produce the offspring
                                    with no constraints.</p>
                                <b style="color: black;">Objective for minimization:</b> <img
                                    src="images/min-father-4.png"
                                    width="90em">
                            </div>
                        </div>
                    </div>
                </div>
            </div>
        </div>

    </div>
    <footer class="site-footer">

        <!--<span class="site-footer-credits">This page was generated by <a href="https://pages.github.com">GitHub Pages</a> using the <a href="https://github.com/jasonlong/cayman-theme">Cayman theme</a> by <a href="https://twitter.com/jasonlong">Jason Long</a>.</span>-->
    </footer>

</section>


</body>
</html>
