<!DOCTYPE html>
<html lang="en-us">
<head>
    <meta charset="UTF-8">
    <title>Computational Biology by computational-bio</title>
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <link rel="stylesheet" type="text/css" href="stylesheets/normalize.css" media="screen">
    <link rel="stylesheet" type="text/css" href="stylesheets/bootstrap.css" media="screen">
    <link rel="stylesheet" type="text/css" href="stylesheets/bootstrap-theme.min.css" media="screen">
    <link href='https://fonts.googleapis.com/css?family=Open+Sans:400,700' rel='stylesheet' type='text/css'>
    <link rel="stylesheet" type="text/css" href="stylesheets/stylesheet.css" media="screen">
    <link rel="stylesheet" type="text/css" href="stylesheets/github-light.css" media="screen">
    <script type="text/javascript" src="js/jquery-2.2.0.min.js"></script>
    <script type="text/javascript" src="js/bootstrap.min.js"></script>
    <script type="text/javascript" src="js/npm.js"></script>

    <script type="text/javascript">

        $(document).ready(function(){});

    </script>
</head>
<body>
<section class="page-header">
    <h1 class="project-name">Computational Biology</h1>
    <h2 class="project-tagline">Computational Biology Project</h2>
</section>

<section class="main-content">
    <!--<h1 style="text-align: center;">
        <a id="coming-soon" class="anchor" href="#coming-soon" aria-hidden="true"><span aria-hidden="true"
                                                                                        class="octicon octicon-link"></span></a><strong>Coming
        Soon</strong>
    </h1>-->

    <div>
        <!-- Nav tabs -->
        <ul class="nav nav-tabs" role="tablist" id="myTabs-main">
            <li role="presentation" class="active"><a href="#sequence-alignment" aria-controls="sequence-alignment" role="tab" data-toggle="tab">Sequence Alignment</a></li>
            <li role="presentation"><a href="#paper-review" aria-controls="paper-review" role="tab" data-toggle="tab">Paper Review</a></li>
        </ul>

        <!-- Tab panes -->
        <div class="tab-content">
            <div role="tabpanel" class="tab-pane fade in active" id="sequence-alignment">
                <h1 style="text-align: center; padding-top: 10px;">
                    <a id="motivation" class="anchor" href="#motivation" aria-hidden="true">
                        <span aria-hidden="true" class="octicon octicon-link"></span></a>
                    <strong>Motivation</strong>
                </h1>

                <p class="text-style">Aligning nucleotide or amino acid sequences is a very common procedure in HIV computational

                    analysis. An accurate alignment is the first step in making a proper and correct analysis of HIV datasets.

                    Sequence alignments are essential for phylogenetic analysis tracing the epidemiology of HIV, but also

                    for interpretations of drug resistance and data mining efforts, where correctly positioning nucleotides or

                    amino acids of different strains with respect to each other is pivotal.</p>

                <hr>

                <h1 style="text-align: center;">
                    <a id="objective" class="anchor" href="#objective" aria-hidden="true">
                        <span aria-hidden="true" class="octicon octicon-link"></span></a>
                    <strong>Objective</strong>
                </h1>

                <p class="text-style">The strains of HIV-1 can be classified into three groups: the "major" group M, the "outlier"
                    group O and

                    the "new" group N. More than 90% of HIV-1 infections belong to HIV-1 group M. Group M there are

                    known to be at least nine genetically distinct subtypes (or clades) of HIV-1. These are subtypes A, B, C, D,

                    F, G, H, J and K.

                    <br><br>Our objective here is to compare <b>HIV-1 A</b> and <b>HIV-1 B</b> virus by determining the similarity
                    between
                    the

                    two.</p>

                <div>
                    <img src="images/pic1.jpg">
                </div>

                <hr>

                <h1 style="text-align: center;">
                    <a id="method" class="anchor" href="#method" aria-hidden="true">
                        <span aria-hidden="true" class="octicon octicon-link"></span></a>
                    <strong>Method</strong>
                </h1>

                <p class="text-style">We selected two sequence of HIV-1 virus, <b>A</b> and <b>B</b> from HIV Sequence Database.</p>

                <br>
                <div>
                    <!-- Nav tabs -->
                    <ul class="nav nav-tabs" role="tablist" id="myTabs">
                        <li role="presentation" class="active"><a href="#hiv-1A" aria-controls="hiv-1A" role="tab" data-toggle="tab">HIV-1 A</a></li>
                        <li role="presentation"><a href="#hiv-1B" aria-controls="hiv-1B" role="tab" data-toggle="tab">HIV-1 B</a></li>
                    </ul>

                    <!-- Tab panes -->
                    <div class="tab-content">
                        <div role="tabpanel" class="tab-pane fade in active" id="hiv-1A">
                            <p class="hiv-sequence">
                                TGGAAGGGTTAATTTACTCCAGGAGAAGACAAGAAATCCTTGATCTGTGGGTCTACCACACACAAGGCTTCTTCCCTGATTGGCAGGAATACACACCAGGGCCAGGGGTTAGATACCCATTAACATTTGGATGGTGCTTCAAGCTAGTACCAGTTGATCCAGATGAGGTAGAGGGGGCTACTGAGGGAGAGAACAACAGCCTGTTACACCCGATATGTCAACATGGAATGGATGATGAGGAGAAAGAAGTATTAAAGTGGAAGTTTGACAGCCGCCTGGCACTAAAACACAGAGCCCAAGAGATGCATCCGGAGTTCTACAAGAACTGAAGAACTGCTGACACAGAAGTTGCTGACAGGGACTTTCTGATGGGGACTTTCCAGGGGAGGTGTGGTTTGGGCGGAGTTGGGGAGTGGCTAACCCTCAGATGCTGCATATAAGCAGCTGCTTCTCGCTTGTACTGGGTCTCTCTTGTTAGACCAGATCGAGCCTGGGAGCTCTCTGGCTAGCTAGGGAACCCACTGCTTAAGCCTCAATAAAGCTTGCCTTGAGTGCTTCAAGTAGTGTGTGCCCATCTGTTGTGACTCTGGTAACTAGAGATCCCTCAGACCTCTTTAGTCAGTGTGGAAAATCTCTAGCA
                            </p>
                        </div>
                        <div role="tabpanel" class="tab-pane fade" id="hiv-1B">
                            <p class="hiv-sequence">
                                TGGATGGGCTAATTTACTCCCAAAAAAGACAAGATATCCTTGATTTGTGGGTCTACCACACACAAGGCTACTTCCCTGATTGGCAGAACTACACACCAGGGCCAGGGACCAGATTTCCACTGACCTTTGGATGGTGCTTCAAGCTAGTACCAGTTGAGCCAGAGAAGGTAGAGGAGGCTAATGAAGGAGAGAACAACTGCTTGTTACACCCTATGAGCCTGCATGGGATGGAGGACCCGGAGAAAGAAGTGTTAGTATGGAAGTTTGACAGCACCCTAGCATTCCATCACGTGGCCCGAGAGCTGCATCCGGAGTACTACAGAGACTGCTGACATTGAGCTTTCTACAAGGGACTTTCCGCTGGGGACTTTCCAGGGAGGCGTGGCCTGGGCGGGACTGGGGAGTGGCGAGCCCTCAGATGCTGCATATAAGCAGCTGCTTTTTGCCTGTACTGGGTCTCTCTGGTTAGACCAGATCTGAGCCTGGGAGCTCTCTGGCTAACTAGGGAACCCACTGCTTAAGCCTCAATAAAGCTTGCCTTGAGTGCTTCAAGTAGTGTGTGCCCGTCTGTTGTGTGACTCTGGTAACTAGAGATCCCTCAGACCCTTTTAGTCAGTGTGGAAAATCTCTAGCA
                            </p>
                        </div>
                    </div>

                </div>
                <br><br>

                <p class="text-style">
                    We compared the DNA strands of HIV-1A and HIV-1B using two different techniques â€“
                </p>

                <div class="panel-group" id="accordion" role="tablist" aria-multiselectable="true">
                    <div class="panel panel-default">
                        <div class="panel-heading" role="tab" id="headingOne">
                            <h4 class="panel-title">
                                <a role="button" data-toggle="collapse" data-parent="#accordion" href="#collapseOne" aria-expanded="true" aria-controls="collapseOne">
                                    1. Multiple Sequence Alignment (MSA)
                                </a>
                            </h4>
                        </div>
                        <div id="collapseOne" class="panel-collapse collapse in" role="tabpanel" aria-labelledby="headingOne">
                            <div class="panel-body">
                                <b> Multiple Sequence Alignment (MSA)</b> is generally the alignment of three or more biological

                                sequences (protein or nucleic acid) of similar length. From the output, homology can be inferred

                                and the evolutionary relationships between the sequences studied.

                                <p>Tool :
                                    <a href="http://www.ebi.ac.uk/Tools/msa/clustalo/">Cluster Omega</a>
                                </p><hr>
                                <h3 style="text-align: center;">
                                    <a id="output-1" class="anchor" href="#output" aria-hidden="true">
                                        <span aria-hidden="true" class="octicon octicon-link"></span></a>
                                    <strong>Output</strong>
                                </h3>
                                <div>
                                    <h4>Sequence Alignment using cluster omega</h4>
                                    <img src="images/Omega1.jpg">
                                </div>
                                <div class="row">
                                    <div class="col-md-6">
                                        <table class="table-striped">
                                            <tbody>
                                            <tr>
                                                <td>Gaps</td>
                                                <td>12</td>
                                            </tr>
                                            <tr>
                                                <td>Mismatches</td>
                                                <td>87</td>
                                            </tr>
                                            <tr>
                                                <td>Similarity</td>
                                                <td>544</td>
                                            </tr>
                                            </tbody>
                                        </table>
                                    </div>
                                    <div class="col-md-6">
                                        <img src="images/Omega-2.jpg">
                                    </div>
                                </div>
                            </div>
                        </div>
                    </div>
                    <div class="panel panel-default">
                        <div class="panel-heading" role="tab" id="headingTwo">
                            <h4 class="panel-title">
                                <a class="collapsed" role="button" data-toggle="collapse" data-parent="#accordion" href="#collapseTwo" aria-expanded="false" aria-controls="collapseTwo">
                                    2. Pair-wise sequence Alignment
                                </a>
                            </h4>
                        </div>
                        <div id="collapseTwo" class="panel-collapse collapse" role="tabpanel" aria-labelledby="headingTwo">
                            <div class="panel-body">
                                <b>Pair-wise sequence Alignment</b> is used to identify regions of similarity that may indicate

                                functional, structural and/or evolutionary relationships between two biological sequences

                                (protein or nucleic acid).
                                <p>Tool :
                                    <a href="http://www.ebi.ac.uk/Tools/psa/emboss_needle/nucleotide.html">Needle</a>
                                </p>
                                <hr>
                                <h3 style="text-align: center;">
                                    <a id="output-2" class="anchor" href="#output" aria-hidden="true">
                                        <span aria-hidden="true" class="octicon octicon-link"></span></a>
                                    <strong>Output</strong>
                                </h3>
                                <div>
                                    <h4>Pairwise Alignment using needle</h4>
                                    <img src="images/Needle2.jpg">
                                    <img src="images/Needle1.jpg">
                                </div>
                            </div>
                        </div>
                    </div>
                </div>
                <br>

                <hr>

                <div style="clear: both;"></div>
                <br>

                <h1 style="text-align: center;">
                    <a id="conclusion" class="anchor" href="#conclusion" aria-hidden="true">
                        <span aria-hidden="true" class="octicon octicon-link"></span></a>
                    <strong>Conclusion</strong>
                </h1>

                <p class="text-style">
                    Even though HIV1A and B originate from different regions viz. Subtype A is common in West Africa while Subtype B
                    is the dominant form in Europe, the Americas, Japan, Thailand, and Australia, The sequence alignment results
                    show a similarity of <b>85-86%</b> position with <b>12-16</b> gaps and <b>~82-87</b> mismatches.
                </p>

                <hr>

                <!--<h1 style="text-align: center;">
                    <a id="future-work" class="anchor" href="#future-work" aria-hidden="true">
                        <span aria-hidden="true" class="octicon octicon-link"></span></a>
                    <strong>Future Work</strong>
                </h1>

                <hr>-->

                <h1 style="text-align: center;">
                    <a id="bibliography" class="anchor" href="#bibliography" aria-hidden="true">
                        <span aria-hidden="true" class="octicon octicon-link"></span></a>
                    <strong>Bibliography</strong>
                </h1>

                <ul class="hiv-sequence">
                    <li>Sequence Alignment in HIV Computational Analysis [
                        http://www.hiv.lanl.gov/content/sequence/HIV/COMPENDIUM/2006_7/abecasis.pdf ]
                    </li>
                    <li>BFCounter Software [ https://github.com/pmelsted/BFCounter/ ]</li>
                </ul>

                <hr>

                <h2 style="text-align: center;">
                    <a id="team" class="anchor" href="#team" aria-hidden="true">
                        <span aria-hidden="true" class="octicon octicon-link"></span></a>
                    Contributors
                </h2>

                <div class="contributors">
                    <div>
                        <a href="https://www.linkedin.com/in/ashwinbansod" class="user-mention">
                            <img src="images/Ashwin.png">
                            <br>Ashwin Bansod</a>
                        <!--<p class="contributors-description">
                            Computer Science graduate student at University of Illinois at Chicago with over 3.8 years of experience
                            in
                            software development, product release, requirement gathering, client interaction. Experience in
                            financial
                            services, stock markets developing low latency high throughput trading system.-->
                    </div>
                    <div>
                        <a href="https://www.linkedin.com/in/krutarthjoshi18" class="user-mention">
                            <img src="images/Krutarth.png">
                            <br>Krutarth Joshi</a>
                        <!--<p class="contributors-description">
                            Pursuing Masters of Science in Computer Science at University of Illinois at Chicago with specialization
                            in
                            Machine Learning and Data Science.
                            Researched various supervised machine learning techniques, and applied them to
                            create an alarm generation system to detect and hence, predict and prevent bearing failures in
                            aero-derivative engines.-->
                    </div>
                    <div>
                        <a href="https://www.linkedin.com/in/kushagrathapar" class="user-mention">
                            <img src="images/Kushagra.png">
                            <br>Kushagra Thapar</a>
                        <!--<p class="contributors-description">
                            Software Developer with 3 years of experience in web application development, web services, database
                            designs, designing product architecture from scratch.
                            Currently pursuing Masters of Science in Computer Science from University of Illinois at Chicago with
                            specialization in Artificial Intelligence-->
                    </div>
                </div>
            </div>
            <div role="tabpanel" class="tab-pane fade" id="paper-review">
                <h2 style="text-align: center; padding-top: 20px">An Exact Algorithm to Compute the DCJ Distance for Genomes with Duplicate Genes</h2>
                <h3 style="text-align: center">
                    Mingfu Shao, Yu Lin, and Bernard Moret
                        <h4 style="text-align: center; margin-top: 1em">
                            Laboratory for Computational Biology and Bioinformatics, EPFL, Lausanne, Switzerland<br>
                                Department of Computer Science and Engineering, University of California, San Diego,
                                La Jolla, California<br>
                            {mingfu.shao,yu.lin,bernard.moret}@epfl.ch
                        </h4>
                </h3>
                <p class="text-style">
                    <br>The double-cut-and-join (DCJ) model has been used to calculate edit distance between two genomes for quite a long time. Optimizing edit distance problem forms an integral part to solve basic problems for genome evolution. Shao, Lin, and Moret proposes an Integer Linear Programming approach to tackle a specific case of genomes i.e. genomes with duplicate genes. This problem can be solved in linear time for genomes without duplicate genes but has been proved NP-Hard for genomes with duplicate genes by a reduction from NP-Hard problem of Breakpoint Graph Decomposition.
                    <br><br>The authors define this problem as to find a valid bijection between two genomes with duplicate gene content that minimizes the DCJ distance between them. When there is a consistent decomposition with <i>c</i> cycles and <i>o</i> odd-length paths, DCJ operations needed to transform genome G1 to G2 is given by <i>(|V|/4 âˆ’ c âˆ’ o/2)</i>. Hence to minimize total DCJ operations, maximize <i>(c + o/2)</i> over the space of all consistent decompositions. This reduces the problem to <i>maximum cycle decomposition problem</i> which is further formulated as an Integer Linear Program. The ILP formulation has O|E| variables, O|E| constraints and maximizes the number of cycles.
                    <br><br>Authors have devised a preprocessing algorithm to reduce complexity while preserving optimality of DCJ distance computing algorithm. The combined approach is crucial for comparing genomes with duplicate genes as they are commonly observed in most species.
                    <br><br>Authors claim a bold assumption that after a speciation event, only DCJ operations are involved. This is unrealistic and is only used to simplify the problem.
                </p>
            </div>
        </div>

    </div>
    <footer class="site-footer">

        <!--<span class="site-footer-credits">This page was generated by <a href="https://pages.github.com">GitHub Pages</a> using the <a href="https://github.com/jasonlong/cayman-theme">Cayman theme</a> by <a href="https://twitter.com/jasonlong">Jason Long</a>.</span>-->
    </footer>

</section>


</body>
</html>
